### PR TITLE
Refactors Subpop labeling

### LINKED ISSUE #2984: Should move sub-population labeling into metric config
many of our topics are only studying a specific sub-population, e.g. HIV PREP is 16+, HIV Stigma is 18+, all AHR are "adults", etc. 

For some of these we manually run `generateSubtitle()` which conditionally hard codes in the sub-population and returns it as a subtitle for each card. This is good, but many are missing and it's pretty hard to read logic that detects and constructs the string.

We should refactor, specifying the subpopulation within the MetricConfig per DataTypeId. That way, the `generateSubtitle()` function would just be responsible for loading the sub-pop string from the config, and then returning the combined focus group, such as `Ages 13+` or `Black Women, Ages 13+`, etc. 

We should address #1907 which is the same idea, but including the "all" specifier on our chart group labels as well (not just on the subtitle of the card).

### LINKED ISSUE #3220: Age sub pops wrong on STIGMA
**Describe the bug**
one card says 13+ and one says 16+

completing #2984 will simplify this logic and make it more clear where the subtitle group names are coming from and hopefully prevent issues like this again

**To Reproduce**
1. Go to https://healthequitytracker.org/exploredata?mls=1.hiv_prep-3.00&group1=All
2. ages are different on comparison bar chart and table chart

**Expected behavior**
all PREP uses the same sub age group and should show that accurately on the page

**Screenshots**

Screenshot 2024-05-02 at 10 41 22 AM

### PR DESCRIPTION
**Pull request by benhammondmusic**
# Description and Motivation

- closes #2984 
- fixes #3220 
- only renders wisqars missing single race alert when its race breakdown
- adds subtitles with specific subpopulation to time-series cards
- rips out half-backed "altLabel" concept to relabel demographic groups right before sending to Vega; instead we now show the overall subpopulation (like `Ages 13+`) in the card subtitle
- updates some variable names in the wisqars_youth provider, adds TODOs for backend clarifying metricId changes
- adds subpopulation counts for CAWP maps to show total number of leg. in the geo context breadcrumbs
- has phrma topics use specific medication denominator population, not medicare ben. in general
- reduces spacing on the "did you know" voter engagement info box

## Has this been tested? How?

- updates vitests, passing
- e2e passing

## Screenshots (if appropriate)

Screenshot 2024-05-02 at 10 41 22 AM
Screenshot 2024-05-02 at 11 25 25 PM
Screenshot 2024-05-02 at 11 26 14 PM
Screenshot 2024-05-02 at 11 28 23 PM
Screenshot 2024-05-02 at 11 29 26 PM

## Types of changes

(leave all that apply)

- Bug fix
- New content or feature
- Refactor / chore

### CODE REVIEW

**Review by eriwarr — CHANGES_REQUESTED** [review]
It looks great!! I made a few suggestions, including removing the commented code and updating the sub-population label from `18+` to `Ages 18+` for Medicare adherence.

File: frontend/src/cards/ui/GeoContext.tsx [hunk 1]
```diff
@@ -49,18 +51,23 @@ const POP_MISSING_VALUE = 'unavailable'
...
+  if (!subPopConfig?.rateDenominatorMetric) return ''
   const allRow = subPopulationData.find((row) => row[demographicType] === ALL)
   const popAllCount: string =
-    allRow?.[subPopConfig.metricId]?.toLocaleString() ?? POP_MISSING_VALUE
-  return `${subPopConfig.shortLabel}: ${popAllCount}`
+    allRow?.[subPopConfig.rateDenominatorMetric?.metricId]?.toLocaleString('en-US', { maximumFractionDigits: 0 }) ?? POP_MISSING_VALUE
+
+  const combinedSubPop = [dataTypeConfig.otherSubPopulationLabel, dataTypeConfig.ageSubPopulationLabel].filter(Boolean).join(', ')
+
+  return `Total ${combinedSubPop ? combinedSubPop + ' ' : ''}population: ${popAllCount}${subPopulationSourceLabel ? ' (from ' + subPopulationSourceLabel + ')' : ''}`
```
[the remark this answers is not in the document]
Comment (benhammondmusic) [reply frontend/src/cards/ui/GeoContext.tsx:72]: Screenshot 2024-05-03 at 11 26 16 AM
Screenshot 2024-05-03 at 11 26 24 AM
Screenshot 2024-05-03 at 11 26 41 AM
Screenshot 2024-05-03 at 11 26 52 AM
Screenshot 2024-05-03 at 11 30 39 AM
Screenshot 2024-05-03 at 11 32 23 AM
Screenshot 2024-05-03 at 11 32 41 AM
Screenshot 2024-05-03 at 11 32 50 AM
Screenshot 2024-05-03 at 11 33 00 AM
Screenshot 2024-05-03 at 11 33 05 AM
Screenshot 2024-05-03 at 11 33 31 AM

**Comment by benhammondmusic** [comment]
Screenshot 2024-05-03 at 11 26 16 AM
Screenshot 2024-05-03 at 11 26 24 AM
Screenshot 2024-05-03 at 11 26 41 AM
Screenshot 2024-05-03 at 11 26 52 AM
Screenshot 2024-05-03 at 11 30 39 AM
Screenshot 2024-05-03 at 11 32 23 AM
Screenshot 2024-05-03 at 11 32 41 AM
Screenshot 2024-05-03 at 11 32 50 AM
Screenshot 2024-05-03 at 11 33 00 AM
Screenshot 2024-05-03 at 11 33 05 AM
Screenshot 2024-05-03 at 11 33 31 AM

**Review by eriwarr — APPROVED** [review]
Awesome!🤘🏿

**Pull request merged**